account_check_deposit/models/account_deposit.py

```diff
@@ -0,0 +1,263 @@
+            currency_none_same_company_id = False
+            if deposit.company_id.currency_id != deposit.currency_id:
+                currency_none_same_company_id = deposit.currency_id.id
+            for line in deposit.check_payment_ids:
+                count += 1
+                if currency_none_same_company_id:
+                    total += line.amount_currency
+                else:
+                    total += line.debit
+            if deposit.move_id:
+                for line in deposit.move_id.line_ids:
+                    if line.debit > 0 and line.reconciled:
+                        reconcile = True
+            deposit.total_amount = total
+            deposit.is_reconcile = reconcile
+            deposit.currency_none_same_company_id =\
+                currency_none_same_company_id
+            deposit.check_count = count
+
+    name = fields.Char(string='Name', size=64, readonly=True, default='/')
+    check_payment_ids = fields.One2many(
+        'account.move.line', 'check_deposit_id', string='Check Payments',
+        states={'done': [('readonly', '=', True)]})
+    deposit_date = fields.Date(
+        string='Deposit Date', required=True,
+        states={'done': [('readonly', '=', True)]},
+        default=fields.Date.context_today)
+    journal_id = fields.Many2one(
+        'account.journal', string='Journal',
+        domain=[('type', '=', 'bank'), ('bank_account_id', '=', False)],
+        required=True, states={'done': [('readonly', '=', True)]})
+    journal_default_account_id = fields.Many2one(
+        'account.account', related='journal_id.default_debit_account_id',
+        string='Default Debit Account of the Journal', readonly=True)
+    currency_id = fields.Many2one(
+        'res.currency', string='Currency', required=True,
+        states={'done': [('readonly', '=', True)]})
+    currency_none_same_company_id = fields.Many2one(
+        'res.currency', compute='_compute_check_deposit',
+        string='Currency (False if same as company)')
+    state = fields.Selection(
+        [
+            ('draft', 'Draft'),
+            ('done', 'Done'),
+        ], string='Status', default='draft', readonly=True)
+    move_id = fields.Many2one(
+        'account.move', string='Journal Entry', readonly=True)
+    bank_journal_id = fields.Many2one(
+        'account.journal', string='Bank Account', required=True,
+        domain="[('company_id', '=', company_id), ('type', '=', 'bank'), "
+        "('bank_account_id', '!=', False)]",
+        states={'done': [('readonly', '=', True)]})
+    line_ids = fields.One2many(
+        'account.move.line', related='move_id.line_ids',
+        string='Lines', readonly=True)
+    company_id = fields.Many2one(
+        'res.company', string='Company', required=True,
+        states={'done': [('readonly', '=', True)]},
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.check.deposit'))
+    total_amount = fields.Float(
```

Comment:
same here or others `compute`

Reply:
@nikul-serpentcs Thanks for your review. And won't add this kind of refactoring in this migration pull request since it's just cosmetic changes. Feel free to open a refactoring PR after the merge.